cpp/test/sparse/preprocess_csr.cu

```diff
@@ -0,0 +1,188 @@
+                                             columns_nnz.view(),
+                                             values_nnz.view(),
+                                             num_rows);
+    auto rows_csr = raft::make_device_vector<int, int64_t>(handle, non_dupe_nnz_count);
+    raft::sparse::convert::sorted_coo_to_csr(
+      rows_nnz.data_handle(), int(rows_nnz.size()), rows_csr.data_handle(), num_rows, stream);
+
+    auto csr_struct_view = raft::make_device_compressed_structure_view(rows_csr.data_handle(),
+                                                                       columns_nnz.data_handle(),
+                                                                       num_rows,
+                                                                       num_cols,
+                                                                       int(values_nnz.size()));
+    auto c_matrix =
+      raft::make_device_csr_matrix<Type_f, Index_, Index_, Index_>(handle, csr_struct_view);
+
+    raft::update_device<Type_f>(
+      c_matrix.view().get_elements().data(), values_nnz.data_handle(), values_nnz.size(), stream);
+
+    auto result     = raft::make_device_vector<Type_f, int64_t>(handle, values_nnz.size());
+    auto bm25_vals  = raft::make_device_vector<Type_f, int64_t>(handle, values_nnz.size());
+    auto tfidf_vals = raft::make_device_vector<Type_f, int64_t>(handle, values_nnz.size());
+
+    if (bm25_on) {
+      sparse::matrix::encode_bm25<Index_, Type_f>(handle, c_matrix.view(), result.view());
+      calc_tfidf_bm25<Index_, Type_f>(handle, c_matrix.view(), bm25_vals.view());
+      ASSERT_TRUE(raft::devArrMatch<Type_f>(bm25_vals.data_handle(),
+                                            result.data_handle(),
+                                            result.size(),
+                                            raft::CompareApprox<Type_f>(2e-5),
+                                            stream));
+    } else {
+      sparse::matrix::encode_tfidf<Index_, Type_f>(handle, c_matrix.view(), result.view());
+      calc_tfidf_bm25<Index_, Type_f>(handle, c_matrix.view(), tfidf_vals.view(), true);
+      ASSERT_TRUE(raft::devArrMatch<Type_f>(tfidf_vals.data_handle(),
+                                            result.data_handle(),
+                                            result.size(),
+                                            raft::CompareApprox<Type_f>(2e-5),
+                                            stream));
+    }
+
+    RAFT_CUDA_TRY(cudaStreamSynchronize(stream));
+  }
+
+ protected:
+  raft::resources handle;
+  cudaStream_t stream;
+
+  SparsePreprocessInputs<Type_f, Index_> params;
+};
+
+using SparsePreprocessTfidfCsr = SparsePreprocessCSR<float, int>;
+TEST_P(SparsePreprocessTfidfCsr, Result) { Run(false); }
+
+using SparsePreprocessBm25Csr = SparsePreprocessCSR<float, int>;
+TEST_P(SparsePreprocessBm25Csr, Result) { Run(true); }
+
+const std::vector<SparsePreprocessInputs<float, int>> sparse_preprocess_inputs = {
```

Comment:
To be conservative and ensure that there are no surprises after merging, it is best to add some use cases for larger matrices.

Reply:
larger random matrices is difficult to ensure, because rmat currently makes many duplicates during edge creation. This results in much smaller than anticipated number of edges. I think in its current form it would be misleading. But I can definitely pass much bigger parameters to RMAT. I dont think the end result will be what we expect. We need to first create a function that creates an RMAT and then removes duplicates and keeps looping through this logic until we get a set of edges of the desired amount that have no duplicates. This is outside of the purview of this PR, IMO. How do you feel about it @cjnolet?